src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -28,219 +86,1018 @@ const ActorInit En_Ik_InitVars = {
+    this->actor.params = IK_GET_ARMOR_TYPE(&this->actor);
+    this->actor.params--;
+
+    Effect_Add(globalCtx, &this->effectIndex, EFFECT_BLURE2, 0, 0, &sBlureInit);
+    if (!sDisplayListDesegmented) {
+
+        for (i = 0; i < ARRAY_COUNT(sIronKnuckleArmorType); i++) {
+            sIronKnuckleArmorType[i][IK_TYPE_SILVER - 1] = Lib_SegmentedToVirtual(sIronKnuckleArmorType[i][0]);
+            sIronKnuckleArmorType[i][IK_TYPE_BLACK - 1] = Lib_SegmentedToVirtual(sIronKnuckleArmorType[i][1]);
+            sIronKnuckleArmorType[i][IK_TYPE_WHITE - 1] = Lib_SegmentedToVirtual(sIronKnuckleArmorType[i][2]);
+        }
+        sDisplayListDesegmented = true;
+    }
+    EnIk_SetupIdle(this);
+}
+
+void EnIk_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnIk* this = THIS;
+
+    Collider_DestroyTris(globalCtx, &this->colliderTris);
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+    Collider_DestroyQuad(globalCtx, &this->colliderQuad);
+    Effect_Destroy(globalCtx, this->effectIndex);
+}
+
+void EnIk_Freeze(EnIk* this) {
+    this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX;
+    this->drawDmgEffScale = 0.65f;
+    this->drawDmgEffFrozenSteamScale = 97.5f * 0.01f;
+    this->drawDmgEffAlpha = 1.0f;
+    this->timer = 80;
+    this->actor.flags &= ~ACTOR_FLAG_400;
+    Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 80);
+}
+
+void EnIk_Thaw(EnIk* this, GlobalContext* globalCtx) {
+    if (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+        this->drawDmgEffAlpha = 0.0f;
+        Actor_SpawnIceEffects(globalCtx, &this->actor, this->limbPos, ARRAY_COUNT(this->limbPos), 2, 0.3f, 0.2f);
+        this->actor.flags |= ACTOR_FLAG_400;
+    }
+}
+
+void EnIk_HitArmor(EnIk* this, GlobalContext* globalCtx) {
+    this->drawDmgEffAlpha = 4.0f;
+    this->drawDmgEffScale = 0.65f;
+    this->drawDmgEffType = ACTOR_DRAW_DMGEFF_LIGHT_ORBS;
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, this->colliderCylinder.info.bumper.hitPos.x,
+                this->colliderCylinder.info.bumper.hitPos.y, this->colliderCylinder.info.bumper.hitPos.z, 0, 0, 0,
+                CLEAR_TAG_LARGE_LIGHT_RAYS);
+}
+
+s32 EnIk_IsChangingAction(EnIk* this, GlobalContext* globalCtx) {
+    if ((this->drawArmorFlags) && (this->actionFunc != EnIk_Block) &&
+        (func_800BE184(globalCtx, &this->actor, 100.0f, 0x2710, 0x4000, this->actor.shape.rot.y) != 0)) {
+        EnIk_SetupBlock(this);
+        return true;
+    }
+    return false;
+}
+
+s32 EnIk_ChooseAttack(EnIk* this) {
+    s16 yawDiff;
+    s32 phi_a2;
+    s32 absYawDiff;
+
+    if ((this->actor.xzDistToPlayer < 100.0f) && (fabsf(this->actor.playerHeightRel) < 150.0f)) {
+
+        phi_a2 = (this->drawArmorFlags == 0) ? 0xAAA : 0x3FFC;
+        yawDiff = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+        absYawDiff = ABS_ALT(yawDiff);
```

Comment:
```suggestion
        absYawDiff= ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.shape.rot.y));
```
Out of curiosity does this work along with removing `yawDiff`. You might have to mess around with the type of `absYawDiff` and possibly move it higher in the stack.